howto/how-to-set-up-a-blaster.md

```diff
@@ -0,0 +1,331 @@
+Speaker is handled exactly the same as in the saber version, and so are the blades. The main difference is that in your hardware installation you will probably use fixed illumination elements rather than removable ones.
+For exposition purposes we will assume that you will be using a Proffie V2.2 board. Also, you will have two buttons, a WS2812B strip on the weapon barrel, powered by LED 2 (pin 19), plus a Red LED for an accent powered by LED 4 (pin 5). You will connect your Fire button (trigger) to Button1 (pin 21) and your Mode button (selector) to Button2 (pin 23). Optionally, you can connect either a Power button,a Reload or a Clip switch to Button3 (pin 22), and of course an OLED.
+
+# Soundfonts
+You will need at least one blaster soundfont. There are some free soundfonts only a search away.
+
+## List of Names and Effects
+| Filename | Effect |
+|---|---|
+| `bgnauto` | Played when auto fire starts. |
+| `auto` | Played while auto fire is going. |
+| `endauto` | Played when auto fire ends. |
+| `blast` | Is the semi-automatic fire sound. You can have as many as you want. |
+| `boot` | Played when ProffieOS boots up. |
+| `clipin` | Sound made when inserting a clip. |
+| `clipout` | Sound made when dropping a clip. |
+| `empty` | Sound when the weapon is out of rounds. |
+| `font` | Name of the preset. |
+| `full` | Sound made when the weapon is full of ammo. |
+| `hum` | Constant sound looping while not firing. |
+| `jam` | Sound made when the weapon jam.s |
+| `unjam` | Sound made when unjamming the blaster. |
+| `mode` | Sound made when switching mode (OS 6+: when `mdkill`, `mdstun` and/or `mdauto` not present) |
+| `plioff` | Played while retracting the PLI bargraph. | 
+| `plion` | Played while extending the PLI bargraph. |
+| `range` |  Sounds of increasing weapon range/power. Currently not implemented. |
+| `stun` | Firing sound for stun mode. |
+| `reload` | Reloading sound. |
+| `mdkill` | (OS 6+) Sound made when switching to KILL mode |
+| `mdstun` | (OS 6+) Sound made when switching to STUN mode |
+| `mdauto` | (OS 6+) Sound made when switching to AUTO mode |
+
+*If no `mdkill`, `mdstun`, `mdauto` nor `mode` are present Talkie voice speaks selected mode.*
+
+# OLED Animations
+[TODO]
+
+# Configuration
+Then comes the time to configure the board's `config.h`. In the `CONFIG_TOP` section the following defines are important for defining your weapon. The rest might still be valid.
+
+## Required Defines
+
+### NUM_BLADES
+Similarly to the saber version, each "blade" is a string of pixels (or a single one) that will have a styled defined for each soundfont. Remember that any accent, crystal chamber and subblade you define counts towards a different blade.
+
+```cpp
+#define NUM_BLADES 2
```

Comment:
I don't think we need to describe all the standard defines, that makes for a lot of duplication.

You might want to just have a list of defines that are useful for blasters and then a pointer to the CONFIG_TOP section where people can read more about those defines.

Reply:
I've rewrote most of the thing working on all your comments and correcter the little typos that prevented the example from compiling. But I did that on my fork's main, and now I don't know how to push it to this request. I'm using the web version of github, btw.

Reply:
I would know how to do this using "git", but not with the web interface.
If you can't figure out, maybe just start a new PR?